net/openvpn-policy-routing/files/openvpn-policy-routing.init

```diff
@@ -0,0 +1,255 @@
+			fi
+			command="-t mangle -A PREROUTING -j MARK --set-mark $mark/$mark -m comment --comment $PKG_NAME"
+			;;
+		*)
+			output "$__error__: $p_name unknown policy gateway ($gw)!\n"; return;;
+	esac
+	[ -z "$laddr" -a -z "$lport" -a -z "$raddr" -a -z "$rport" ] && output "$__error__: $p_name policy '$comment' missing all IPs/ports!\n" && return
+
+	[ -n "$laddr" ] && command="$command -s $laddr"
+	[ -n "$lport" ] && command="$command -p tcp -m multiport --sport ${lport//-/:}" && lport=":${lport}"
+	[ -n "$raddr" ] && command="$command -d $raddr"
+	[ -n "$rport" ] && command="$command -p tcp -m multiport --dport ${rport//-/:}" && rport=":${rport}"
+
+	if [ "ip_family" == "ipv6" ]; then
+		command="ip6tables $command"
+		[ -z "$laddr" ] && network_get_subnet laddr lan6
+		[ -z "$raddr" ] && raddr'::/0'
+	else
+		command="iptables $command"
+		[ -z "$laddr" ] && network_get_subnet laddr lan
+		[ -z "$raddr" ] && raddr='*.*.*.*'
+	fi
+
+	[ -n "$comment" ] && label="'${comment}' via ${gw} (${laddr}${lport} to ${raddr}${rport})" || label="${laddr}${lport} to ${raddr}${rport} via ${gw}"
+	output "routing $label" 2
+	$command && output " $__ok__\n" 2 || output " $__fail__\n${command}\n" 2
+}
+
+f_dnsmasq(){ local s=1; config_load dhcp; config_foreach process_dnsmasq dnsmasq "$1"; return $s; }
+process_dnsmasq(){
+	local i
+	case $2 in
+		active) [ "$(uci -q get dhcp.$1.port)" == "0" ] || s=0;;
+		ipset)  [ "$(uci -q get dhcp.$1.ipset)" ] && s=0;;
+		output) for i in $(uci -q get dhcp.$1.ipset); do output '.' 2; done;;
+	esac
+}
+
+boot() { ubus -t 30 wait_for network.interface.wan && { rc_procd start_service; rc_procd service_triggers; } || output "$__error__: $p_name failed to settle network interface!\n"; }
+
+start_service() {
+	local wan_gw c=1 ipsl d d_status
+  opr_enabled || return 1
+
+	procd_open_instance
+	procd_set_param stdout 1
+	procd_set_param stderr 1
+	procd_close_instance
+
+	while : ; do network_get_gateway wan_gw wan; [ $c -ge 15 ] && break || let "c+=1"; [ -n "$wan_gw" ] && break || output "$p_name waiting for wan gateway...\n"; sleep 2; network_flush_cache; done
+	[ -z "$wan_gw" ] && output "$__error__: $p_name could not discover wan gateway!\n" && exit 1
+
+	iptables_reset; { modprobe xt_set; modprobe ip_set; modprobe ip_set_hash_ip; } >/dev/null 2>&1;
+	config_load network; config_foreach process_interfaces interface create;
+	config_load openvpn-policy-routing; config_foreach process_policy policy; config_foreach process_policy policy-ipv6 "ipv6";
+	if [ uci -q get openvpn-policy-routing.@domain-policy[] >/dev/null 2>&1 ]; then
+		output "routing domain-based policies" 2;
+		config_foreach process_domain_policy domain-policy
+		[ -z "$d_status" ] && output " $__ok__\n" 2 || output " $__fail__\n$__error__:$d_status\n" 2
+	elif f_dnsmasq "ipset"; then
+		output "routing domain-based policies with dnsmasq" 2;
+		if [ ! -e /var/lock/opkg.lock ] && ! opkg list_installed | grep 'dnsmasq-full' >/dev/null 2>&1 ; then
```

Comment:
Thank you! I was sure I saw it somewhere, but I've tried dnsmasq --help and didn't gather I should have tried -v from it.